src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -2,23 +2,207 @@
+                    {
+                        reference = rightRef;
+                        constant = leftConst;
+                    }
+                    else continue;
+
+                    refParam = singleParameters.First(p => p.Name == reference.ParameterName);
+                    countParameters.Add(refParam);
+                }
+                else continue;
+
+                if (constant.Value == vectorSize)
+                {
+                    CSType vector = new CSType(typeName, baseType.Constant);
+                    singleParameters[i] = parameter with
+                    {
+                        Type = new CSRef(ptr.Constant || baseType.Constant ? CSRef.Type.In : CSRef.Type.Ref,
+                            vector),
+                        Length = null
+                    };
+                    spanParameters[i] = parameter with { Type = new CSSpan(vector, ptr.Constant || baseType.Constant), Length = null };
+                    arrayParameters[i] = parameter with { Type = new CSArray(vector, ptr.Constant || baseType.Constant), Length = null };
+                    nameTable.Rename(parameter, parameter.Name + "_ptr");
+                    overloadedParams.Add((parameter, singleParameters[i], firstElem));
+                }
+            }
+
+            if (overloadedParams.Count == 0)
+            {
+                newOverloads = null;
+                return false;
+            }
+
+            string overloadName = overload.OverloadName.Remove(m.Index + 2, 1);
+            newOverloads = new List<Overload>()
+            {
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = singleParameters.Except(countParameters).ToArray(),
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SingleVectorLayer(countParameters, overloadedParams)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = spanParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameters, overloadedParams)
+                },
+                overload with
+                {
+                    OverloadName = overloadName,
+                    InputParameters = arrayParameters,
+                    NameTable = nameTable,
+                    NestedOverload = overload,
+                    MarshalLayerToNested = new SpanArrayVectorLayer(countParameters, overloadedParams)
+                },
+            };
+            return true;
+        }
+
+        public record SingleVectorLayer(
+            List<Parameter> CountParameters,
+            List<(Parameter ptrParam, Parameter vectorParam, string firstElem)> OverloadedParameters) : IOverloadLayer
```

Comment:
There names should be PascalCase (`PtrParam`,  `VectorParam`, `FirstElem`) as they are public names of a struct.